spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -158,6 +163,26 @@ public void filter(Predicate[] predicates) {
     }
   }
 
+  protected Map<String, DeleteFileSet> dataToFileScopedDeletes() {
```

Comment:
Worth double checking that this has the most filtered down mapping if runtime filtering is performed. I think the write should only be performed after runtime filtering is performed so this should already be happening (so the tasks list is reset) but if it's somehow possible that we compute this mapping during the write and then after that more filters are applied then this wouldn't always be the most filtered down mapping. But that affects more on memory consumption and not correctness.